coveralls/reporter.py

```diff
@@ -101,7 +189,14 @@ def get_arcs(analysis):
         if not analysis.has_arcs():
             return None
 
-        branch_lines = analysis.branch_lines()
+        if not hasattr(analysis, 'branch_lines'):
+            # N.B. switching to the public method analysis.missing_branch_arcs
```

Comment:
@nedbat is there anything I missed in the public `coverage v5+` API which would support this? Specifically, its the following block that I couldn't work out how to do without private APIs:
```python
for l1, l2 in analysis.arcs_executed():
    if l1 in branch_lines:
        # SNIP
```

Reply:
@TheKevJames let's talk about this in another place.  Clearly the Coverage public API doesn't give you everything you need yet.  You called out this spot, but looking at the rest of the file, there are many things you are using from coverage.py that I didn't anticipate you would need.

I've added an issue for coverage.py, let's discuss the details there: https://github.com/nedbat/coveragepy/issues/921